### ISSUE TITLE
lint/nursery/useExplicitType massive differences

### ISSUE THREAD

**Issue by GiuseppeIII** [biomejs/biome#5847]
### Environment information

```bash
...
```

### Rule name

lint/nursery/useExplicitType

### Playground link

https://next.biomejs.dev/playground/?lintRules=all&code=YwBvAG4AcwB0ACAAdABlAHMAdAAgAD0AIAAxACAAYQBzACAAYwBvAG4AcwB0AA%3D%3D

### Expected result

`useExplicitType` is now a much different rule. `useExplicitType` used to be (as it says in its documentation) equivalent to https://typescript-eslint.io/rules/explicit-function-return-type/. Now it seems to have massively increased in scope to all variables that don't have a type to them (even if they are defined in a way where the type is very strictly inferred (see playground)) to where even something like this
```
const test = 1 as const;
```
gets caught up. I would appreciate if it would be possible to separate the additions to this rule out as separate rule (or at the very least as an option on the rule) 

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Thank you for your report.

I'm going to close this issue to avoid having too many discussions.

The rule is still a nursery rule and we're also evaluated multiple rules. We would appreciate it if you could get involved in the issue https://github.com/biomejs/biome/issues/2017 and give us some pointers in terms of options/rules.
